@here/harp-mapview/lib/MapView.ts

```diff
@@ -2518,6 +2538,26 @@ export class MapView extends THREE.EventDispatcher {
         return this.m_elevationProvider;
     }
 
+    get shadowsEnabled(): boolean {
+        return this.m_options.enableShadows === true;
+    }
+
+    set shadowsEnabled(enabled: boolean) {
+        if (
+            this.m_renderer.shadowMap === undefined ||
```

Comment:
If 'this.m_renderer.shadowMap === undefined' and `enabled === true` we should create it and not just return ?!

Reply:
I checked the three.js code, it is never undefined, see: https://github.com/mrdoob/three.js/blob/e42922ad88d0ab406fb48b30ccf271336c46336a/src/renderers/WebGLRenderer.js#L314

I will remove this check.

Reply:
I need this check, otherwise the tests fail.

Reply:
which tests?

Reply:
the tests stub the GL renderer:

```
const webGlStub = sandbox
        .stub(THREE, "WebGLRenderer")
        .returns(TestUtils.getWebGLRendererStub(sandbox, clearColorStub));
```